### PR TITLE
Fix botany bugs

### PR DESCRIPTION
**Pull request by Crotalus**
## About the PR
Botany bugfixes:

* Fix MutateInt code so it takes min/max into account 
See: #21085, #25100
* Fix missing gas warning getting stuck on hydroponics tray
* Set correct number of totalBits
* Add division by zero check

### CODE REVIEW

File: Content.Server/Botany/Systems/MutationSystem.cs [hunk 1]
```diff
@@ -186,10 +192,22 @@ private void MutateInt(ref int val, int min, int max, int bits, int totalbits, f
...
+        {
+            val = min;
+            return;
+        }
+
+        // Starting number of bits that are high, between 0 and bits.
+        // In other words, it's val mapped linearly from range [min, max] to range [0, bits], and then rounded.
+        int valInt = (int) MathF.Round((val - min) / (max - min) * bits);
+        // val may be outside the range of min/max due to starting prototype values, so clamp.
+        valInt = Math.Clamp(valInt, 0, bits);
```
Comment (Partmedia) [inline Content.Server/Botany/Systems/MutationSystem.cs:205]: Please eliminate the space after the cast per the [style guidelines](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions). Also consider using `MathHelper.Lerp` which might convey the intent better.

File: Content.Server/Botany/Systems/MutationSystem.cs [hunk 2]
```diff
@@ -186,10 +192,22 @@ private void MutateInt(ref int val, int min, int max, int bits, int totalbits, f
...
+
         // Probability that the bit flip increases n.
-        // The higher the current value is, the lower the probability of increasing value is, and the higher the probability of decreasive it it.
+        // The higher the current value is, the lower the probability of increasing value is, and the higher the probability of decreasing it.
         // In other words, it tends to go to the middle.
-        float probIncrease = 1 - (float)val / bits;
+        float probIncrease = 1 - (float) valInt / bits;
```
Comment (Partmedia) [inline Content.Server/Botany/Systems/MutationSystem.cs:210]: Cast style

**Review by Partmedia — CHANGES_REQUESTED** [review]
Thanks for the fixes. Would you mind looking at the few small nitpicks?

**Comment by AJCM-git** [comment]
@Crotalus are you coming back to this?

File: Content.Server/Botany/Systems/MutationSystem.cs (hunk 1 above)
Comment (Crotalus) [reply Content.Server/Botany/Systems/MutationSystem.cs]: Fixed the spacing

File: Content.Server/Botany/Systems/MutationSystem.cs (hunk 2 above)
Comment (Crotalus) [reply Content.Server/Botany/Systems/MutationSystem.cs]: Fixed

**Comment by Crotalus** [comment]
@AJCM-git @Partmedia 

Sorry for the delay, I'm back now. I rebased it on the latest master.

I'll make a new PR after this that tries to clean up the code more and use MathHelper.Lerp

**Comment by Crotalus** [comment]
@Partmedia Fixed the merge conflicts, we need the annoying gas bug gone 😄

**Comment by slarticodefast** [comment]
I had a look at the math and made a plot of the resulting probabilities to double check this.
Unbenannt
There is a flat part in the curve due to the rounding of the bits, which is a little weird in general, but not your fault. Considering this is now the same as for the floating point mutations this should be fine though. Thanks for the fix!

**Pull request merged**